source/simulator/assembly.cc

```diff
@@ -219,15 +193,35 @@ namespace aspect
             assemblers->advection_system_assembler_on_face_properties[0].need_face_finite_element_evaluation = true;
           }
 
-        if (parameters.use_discontinuous_composition_discretization)
+        if (i > 0 && parameters.use_discontinuous_composition_discretization)
           {
-            for (unsigned int c = 1; c<=introspection.n_compositional_fields; ++c)
-              {
-                assemblers->advection_system_assembler_on_face_properties[c].need_face_material_model_data = true;
-                assemblers->advection_system_assembler_on_face_properties[c].need_face_finite_element_evaluation = true;
-              }
+            // TODO should these only be set when method is fem_field?
+            assemblers->advection_system_assembler_on_face_properties[i].need_face_material_model_data = true;
+            assemblers->advection_system_assembler_on_face_properties[i].need_face_finite_element_evaluation = true;
```

Comment:
Let's do this in a follow-up PR. It is possible that you're right, and the easiest way to test this is to make a PR that makes this change and see if anything breaks :-)